azure_monitor/src/azure_monitor/auto_collection/dependency_metrics.py

```diff
@@ -14,10 +15,12 @@
 
 def dependency_patch(*args, **kwargs) -> None:
     result = ORIGINAL_REQUEST(*args, **kwargs)
-    # We don't want multiple threads updating this at once
-    with _dependency_lock:
-        count = dependency_map.get("count", 0)
-        dependency_map["count"] = count + 1
+    # Only collect request metric if sent from non-exporter thread
+    if context.get_value("suppress_instrumentation") is None:
```

Comment:
Don't we need to set `supress_instrumentation` in the context for exporter threads?

Reply:
We don't need to add it in the exporter, SpanProcessor will add it before calling the exporter

Reply:
It seems only `SimpleExportSpanProcessor` does this? What about the other `SpanProcessors`? And what about surpressing instrumentation for the metrics exporter as well? Also, are we going to ignore dependency calls that happen during the same time as export?

Reply:
BatchExportSpanProcessor also add this, you are totally right this need to be taken care of in OT repo for Metrics, and I'm not sure how we can ensure this applies to all SpanProcessors, maybe a topic for OT SIG meeting

Reply:
Created issue in OT to track this
https://github.com/open-telemetry/opentelemetry-python/issues/524

Reply:
If OT does not have this, any point in having this PR merged? It also does not address the issue of dependency calls being made at the same time the export is being called.

Reply:
Not sure I understand the issue you mention, the point for merging this here is to address the issue for Spans, so we don't send metrics for internal calls to Breeze, the code in Azure repo would not be affected once issue is resolved in OT so once Metrics add correct flag this will be automatically working here. If you think there are big issues with this flag it would be good to bring it up in OT repo because HTTP request ext is heavily relying on this

Reply:
What I am referring to is, this mechanism for using `suppress_instrumentation` does not cover the use case of when a http request call is made DURING exporting (that's not from the exporter itself). With that being said, IF we are still going with the `suppress_instrumentation` route to address this problem, you are correct that we don't have to change anything in this repo. I would add a comment here indicating that it is contigent on changes in the OT repo.